index.js

```diff
@@ -60,7 +60,7 @@ var torrentStream = function (link, opts, cb) {
   var metadata = link.infoBuffer || null
   var infoHash = link.infoHash
 
-  if (!opts) opts = {}
+  opts = Object.assign(opts || {})
```

Comment:
It will still mutate `opts` like that, since it doesn't create a new object -- did you mean `opts = Object.assign({}, opts)`?

Reply:
Yes, I forgot to put the empty object, thanks.